### PR TITLE
codecov coverage uploaded after tests succeed

### PR DESCRIPTION
**Pull request by michaelraczycki**
**What is this PR about?**
Adaptations of tests.yaml requested in #6660 

**Implementation details:
Removed partial codecov upload that was triggered after each subtest, instead added one that is made only In case of all tests passing

----
:books: Documentation preview :books:: https://pymc--6705.org.readthedocs.build/en/6705/

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -488,3 +485,54 @@ jobs:
...
+          # Activate conda environment
+          conda activate pymc-test
+          # Find all coverage reports and merge them into a single file
+          python -m coverage combine --append coverage-*.xml
+
+      - name: Save combined coverage report
+        uses: actions/upload-artifact@v2
+        with:
+          name: coverage
+          path: coverage
```
Comment (ricardoV94) [inline .github/workflows/tests.yml]: Why do we need this extra job? PyTensor doesn't seem to have an equivalent one.

Comment (michaelraczycki) [reply .github/workflows/tests.yml]: pytensor has just one job ('test_ubuntu') that generates the codecov report. in pymc there are many more, that generate partial reports. Adding step that combines them into one final report after all report-generating jobs finish seemed like an easier task

Comment (ricardoV94) [reply .github/workflows/tests.yml]: Okay, but it can still be done in a single job no? Just merge the two steps

Comment (michaelraczycki) [reply .github/workflows/tests.yml]: by two steps do you mean merge-reports and upload coverage?

Comment (ricardoV94) [reply .github/workflows/tests.yml]: Yes, this way there's a bit less clutter in the PR checks

Comment (michaelraczycki) [reply .github/workflows/tests.yml]: sure, should be possible

**Pull request closed without merging**